### PR TITLE
[Feature] filter_size can be a array 

### LINKED ISSUE #317: [feature request] filter_size can be a array instead of one value
My conv filter is not a square matrix. Its height is 1 and weight is 3. 
In the conv function within the utils.py, the input parameter, filter_size, needs to be an array instead of a value. It is rather simple change.

### PR DESCRIPTION
**Pull request by yutingsz**
This update is regrading the issue #317  [feature request] filter_size can be a array instead of integer

closes #317

### CODE REVIEW

File: stable_baselines/a2c/utils.py [hunk 1]
```diff
@@ -102,14 +102,24 @@ def conv(input_tensor, scope, *, n_filters, filter_size, stride,
...
+    or the height and width of filter if the input is a list
     :param stride: (int) The stride of the convolution
     :param pad: (str) The padding type ('VALID' or 'SAME')
     :param init_scale: (int) The initialization scale
     :param data_format: (str) The data format for the convolution weights
     :param one_dim_bias: (bool) If the bias should be one dimentional or not
     :return: (TensorFlow Tensor) 2d convolutional layer
     """
+    if isinstance(filter_size, list) or isinstance(filter_size, tuple):
+        if len(filter_size) == 2:
```
Comment (araffin) [inline stable_baselines/a2c/utils.py]: I would rather do an assert and add a meaningful error message ;)

Comment (yutingsz) [reply stable_baselines/a2c/utils.py]: New code: 
`    if isinstance(filter_size, list) or isinstance(filter_size, tuple):
        assert len(filter_size) == 2, "Filter size needs to have 2 elements, both height and width!"`  
is it better?

File: stable_baselines/a2c/utils.py [hunk 2]
```diff
@@ -102,14 +102,22 @@ def conv(input_tensor, scope, *, n_filters, filter_size, stride,
     :param input_tensor: (TensorFlow Tensor) The input tensor for the convolution
     :param scope: (str) The TensorFlow variable scope
     :param n_filters: (int) The number of filters
-    :param filter_size: (int) The filter size
+    :param filter_size: either (int) or a list or tuple [int, int]. The filter size for the squared matrix,
```
Comment (araffin) [inline stable_baselines/a2c/utils.py]:
```suggestion
    :param filter_size:  (Union[int, [int], tuple<int, int>]) The filter size for the squared kernel matrix,
```

File: stable_baselines/a2c/utils.py [hunk 3]
```diff
@@ -102,14 +102,22 @@ def conv(input_tensor, scope, *, n_filters, filter_size, stride,
...
+    or the height and width of filter if the input is a list
     :param stride: (int) The stride of the convolution
     :param pad: (str) The padding type ('VALID' or 'SAME')
     :param init_scale: (int) The initialization scale
     :param data_format: (str) The data format for the convolution weights
     :param one_dim_bias: (bool) If the bias should be one dimentional or not
     :return: (TensorFlow Tensor) 2d convolutional layer
     """
+    if isinstance(filter_size, list) or isinstance(filter_size, tuple):
+        assert len(filter_size) == 2, "Filter size needs to have 2 elements, both height and width!"
```
Comment (araffin) [inline stable_baselines/a2c/utils.py]:
```suggestion
        assert len(filter_size) == 2, "Filter size must have 2 elements (height, width), {} were given".format(len(filter_size))
```

**Comment by araffin** [comment]
Please add a test that ensure the feature is working and then we can merge your PR ;)

File: tests/test_a2c_conv.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+ENV_ID = 'BreakoutNoFrameskip-v4'
+SEED = 3
+NUM_TIMESTEPS = 500
+NUM_CPU = 2
+
+
+@pytest.mark.parametrize("input1", [4])
+@pytest.mark.parametrize("input2", [(3, 5)])
+def test_conv_kernel(input1, input2):
```
Comment (araffin) [inline tests/test_a2c_conv.py]: Why do you use parametrize here?

File: tests/test_a2c_conv.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    :param input1: (int) for squared kernel matrix
+    :param input2: (int, int) for non-squared kernel matrix
+    """
+    kwargs = {}
+    n_envs = 1
+    n_steps = 2
+    n_batch = n_envs * n_steps
+    scale = False
+    env = VecFrameStack(make_atari_env(env_id=ENV_ID, num_env=n_envs, seed=SEED), 4)
```
Comment (araffin) [inline tests/test_a2c_conv.py]: Why not a single without all the wrappers?

Like `env = gym.make(ENV_ID)`

File: tests/test_a2c_conv.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    graph = tf.Graph()
+    with graph.as_default():
+        _, scaled_images = observation_input(ob_space, n_batch, scale=scale)
+        activ = tf.nn.relu
+        layer_1 = activ(conv(scaled_images, 'c1', n_filters=32, filter_size=input1, stride=4
+                             , init_scale=np.sqrt(2), **kwargs))
+        layer_2 = activ(conv(layer_1, 'c2', n_filters=32, filter_size=input2, stride=4
+                             , init_scale=np.sqrt(2), **kwargs))
+        print(layer_2.shape)
```
Comment (araffin) [inline tests/test_a2c_conv.py]: Please use an assert here to check that the output shape is as expected

File: tests/test_a2c_conv.py (hunk 4 above)
Comment (yutingsz) [reply tests/test_a2c_conv.py]: Is there any drawback to use parametrize? If there is a better way, I am happy to change it.

File: tests/test_a2c_conv.py (hunk 5 above)
Comment (yutingsz) [reply tests/test_a2c_conv.py]: OK, I can use the gym.make.

File: tests/test_a2c_conv.py (hunk 6 above)
Comment (yutingsz) [reply tests/test_a2c_conv.py]: Will put an assert here to check the shape

File: tests/test_a2c_conv.py (hunk 4 above)
Comment (araffin) [reply tests/test_a2c_conv.py]: well, parametrize is useful when you want to test several values..., here you are using only one, so it does not really make sense.

Comment (araffin) [reply tests/test_a2c_conv.py]: also, you should use variable name that are meaningful, (eg filter_size_1 instead of input_1)

**Pull request merged**